### PR TITLE
[Chore]: Technical: add types for editor component

### PR DESCRIPTION
**Pull request by dariaterekhovaae**
Signed-off-by: Daria Terekhova <daria.terekhova@actionengine.com>

### CODE REVIEW

File: src/components/editor/editor.tsx [hunk 1]
```diff
@@ -18,57 +18,67 @@
...
-    };
+interface EditorProps {
+  filters: Filter[];
+  layers: Layer[];
+  datasets: Datasets;
+  editor: {selectedFeature: Feature; mode: string};
+  layersToRender: Record<string, Layer>;
+  index: number;
+  className: string; // ??
+  classnames: string;
```
Comment (heshan0131) [inline src/components/editor/editor.tsx:59]: why need `classnames`

File: src/components/editor/editor.tsx [hunk 2]
```diff
@@ -18,57 +18,67 @@
...
+  isEnabled: boolean;
+  onSelect: (f: Feature | null) => void;
+  onUpdate: (f: Feature[]) => void;
+  onDeleteFeature: (f: Feature) => void;
+  onTogglePolygonFilter: (l: Layer, f: Feature) => void;
+}
 
+export default function EditorFactory(
+  FeatureActionPanel: ComponentType<FeatureActionPanelProps>
+): ComponentType<EditorProps> {
```
Comment (heshan0131) [inline src/components/editor/editor.tsx:71]: should this be `Reac.Component<EditorProps>`?

File: src/components/editor/editor.tsx [hunk 3]
```diff
@@ -18,57 +18,67 @@
...
+  style: CSSProperties;
+  isEnabled: boolean;
+  onSelect: (f: Feature | null) => void;
+  onUpdate: (f: Feature[]) => void;
+  onDeleteFeature: (f: Feature) => void;
+  onTogglePolygonFilter: (l: Layer, f: Feature) => void;
+}
 
+export default function EditorFactory(
+  FeatureActionPanel: ComponentType<FeatureActionPanelProps>
```
Comment (heshan0131) [inline src/components/editor/editor.tsx:70]: should this be `React.FC<FeatureActionPanelProps>`?

File: src/components/editor/feature-action-panel.tsx [hunk 4]
```diff
@@ -37,7 +39,22 @@ const StyledActionsLayer = styled.div`
...
+    y: number;
+  };
+  layers: Layer[];
+  currentFilter?: Filter;
+  onToggleLayer: (l: Layer) => void;
+  onDeleteFeature: () => void;
+  onClose: () => void;
+}
+
+export function PureFeatureActionPanelFactory(): ComponentType<FeatureActionPanelProps> {
```
Comment (heshan0131) [inline src/components/editor/feature-action-panel.tsx:57]: `const FeatureActionPanel: React.FC<FeatureActionPanelProps>`

File: src/components/editor/feature-action-panel.tsx [hunk 5]
```diff
@@ -37,7 +39,22 @@ const StyledActionsLayer = styled.div`
 
 PureFeatureActionPanelFactory.deps = [];
 
-export function PureFeatureActionPanelFactory() {
+export interface FeatureActionPanelProps {
+  className?: string;
+  datasets: Datasets;
+  selectedFeature: Feature;
```
Comment (heshan0131) [inline src/components/editor/feature-action-panel.tsx:45]: `selectedFeature` can be `null`

File: src/components/editor/feature-action-panel.tsx [hunk 6]
```diff
@@ -124,9 +132,9 @@ FeatureActionPanelFactory.deps = PureFeatureActionPanelFactory.deps;
 export default function FeatureActionPanelFactory() {
   const PureFeatureActionPanel = PureFeatureActionPanelFactory();
 
-  const ClickOutsideFeatureActionPanel = props => {
+  const ClickOutsideFeatureActionPanel = (props: FeatureActionPanelProps) => {
```
Comment (heshan0131) [inline src/components/editor/feature-action-panel.tsx:135]: `const ClickOutsideFeatureActionPanel: React.FC<FeatureActionPanelProps>`

Comment (jagerts) [reply src/components/editor/feature-action-panel.tsx:135]: fixed

File: src/components/editor/feature-action-panel.tsx (hunk 4 above)
Comment (jagerts) [reply src/components/editor/feature-action-panel.tsx:57]: fixed

File: src/components/editor/feature-action-panel.tsx (hunk 5 above)
Comment (jagerts) [reply src/components/editor/feature-action-panel.tsx:45]: fixed

File: src/components/editor/editor.tsx (hunk 2 above)
Comment (jagerts) [reply src/components/editor/editor.tsx:71]: replaced by React.ComponentClass<EditorProps>

File: src/components/editor/editor.tsx (hunk 3 above)
Comment (jagerts) [reply src/components/editor/editor.tsx:70]: fixed

File: src/components/editor/editor.tsx (hunk 1 above)
Comment (jagerts) [reply src/components/editor/editor.tsx:59]: replaced

**Comment by btford** [comment]
LGTM

File: src/components/editor/editor.tsx [hunk 7]
```diff
@@ -18,57 +18,65 @@
...
     static defaultProps = {
       clickRadius: DEFAULT_RADIUS
     };
 
+    static displayName = 'Editor';
+
     state = {
       showActions: false,
-      lastPosition: null
+      lastPosition: {x: 0, y: 0}
```
Comment (heshan0131) [inline src/components/editor/editor.tsx:79]: keep this as `null`
in `feature-action-panel` add 
```js
if (!position) {
  return null
}

 return (
      <StyledActionsLayer
        className={classnames('feature-action-panel', className)}
        style={{
          top: `${position.y + LAYOVER_OFFSET}px`,
          left: `${position.x + LAYOVER_OFFSET}px`
        }}
      >
...

```

File: src/components/editor/feature-action-panel.tsx [hunk 8]
```diff
@@ -102,18 +120,8 @@ export function PureFeatureActionPanelFactory() {
...
-    position: PropTypes.object.isRequired,
-    layers: PropTypes.arrayOf(PropTypes.object).isRequired,
-    currentFilter: PropTypes.object,
-    onClose: PropTypes.func.isRequired,
-    onDeleteFeature: PropTypes.func.isRequired
-  };
-
   FeatureActionPanel.defaultProps = {
-    position: {}
+    position: {x: 0, y: 0}
```
Comment (heshan0131) [inline src/components/editor/feature-action-panel.tsx:124]: make default position to be null. and return `null` in render if `position` is `null`

File: src/components/editor/editor.tsx (hunk 7 above)
Comment (jagerts) [reply src/components/editor/editor.tsx]: fixed

File: src/components/editor/feature-action-panel.tsx (hunk 8 above)
Comment (jagerts) [reply src/components/editor/feature-action-panel.tsx]: fixed
also fixed a test for FeatureActionPanel, plz check it out

**Review by heshan0131 — APPROVED** [review]

**Review by heshan0131 — APPROVED** [review]

**Pull request merged**